command/agent/config.go

```diff
@@ -668,22 +668,86 @@ func (r *Resources) CanParseReserved() error {
 	return err
 }
 
+// devModeConfig holds the config for the -dev flag
+type devModeConfig struct {
+	defaultMode bool
+	connectMode bool
+}
+
+// newDevMode parses the optional string value of the -dev flag
+func newDevMode(s string) (*devModeConfig, error) {
+	if s == "" {
+		return nil, nil // no -dev flag
+	}
+	mode := &devModeConfig{}
+	modeFlags := strings.Split(s, ",")
+	for _, modeFlag := range modeFlags {
+		switch modeFlag {
+		case "true": // -dev flag with no params
+			mode.defaultMode = true
+		case "connect":
+			if runtime.GOOS != "linux" {
+				// strictly speaking -dev=connect only binds to the
+				// non-localhost interface, but given its purpose
+				// is to support a feature with network namespaces
+				// we'll return an error here rather than let the agent
+				// come up and fail unexpectedly to run jobs
+				return nil, fmt.Errorf("-dev=connect is only supported on linux")
+			}
+			mode.connectMode = true
+		default:
+			return nil, fmt.Errorf("invalid -dev flag")
+		}
+	}
+	return mode, nil
+}
+
+func (mode *devModeConfig) networkConfig() (string, string) {
+	if runtime.GOOS == "darwin" {
+		return "127.0.0.1", "lo0"
+	}
+	if mode != nil && mode.connectMode {
+		// if we hit either of the unchecked errors here we're in a weird
```

Comment:
I wonder if we can use a go-socketaddr template here. See https://godoc.org/github.com/hashicorp/go-sockaddr/template

The following should work:
```
{{ GetDefaultInterfaces | limit 1 | attr "name" }}
```

Reply:
Oh I didn't know about that library... that's really cool. Definitely would be much nicer!

Reply:
Done. And now I also see that the `socketaddr` command line invocation in the original ticket was referring to this and not some random third-party CLI tool. 😆